### PR TITLE
[Impeller] fix Xcode frame capture.

### PR DESCRIPTION
**Pull request by jonahwilliams**
This fixed things for me locally. Confirmed frame captures work and this block otherwise doesn't fire when running regularly

### CODE REVIEW

**Review by dnfield — APPROVED** [review]
LGTM

File: impeller/renderer/backend/metal/surface_mtl.mm [hunk 1]
```diff
@@ -188,6 +188,15 @@
       ContextMTL::Cast(context.get())->CreateMTLCommandBuffer();
   [command_buffer commit];
   [command_buffer waitUntilScheduled];
+#else
```
Comment (chinmaygarde) [inline impeller/renderer/backend/metal/surface_mtl.mm:191]: Can we just move this into its own method to dry this up? We can move the comments and such to that method too.

```
if (ShouldWaitUntilScheduled()) {
  [command_buffer commit];
  [command_buffer waitUntilSh...];
}
```

**Review by chinmaygarde — APPROVED** [review]
LGTM with nits.

File: impeller/renderer/backend/metal/surface_mtl.mm (hunk 1 above)
Comment (jonahwilliams) [reply impeller/renderer/backend/metal/surface_mtl.mm]: Done

**Pull request merged**